crates/optimism/node/src/txpool.rs

```diff
@@ -348,34 +363,100 @@ where
+
+            // If the EVM is configured, enable interop tx execution.
+            if let Some(evm_cfg) = &self.evm {
+                // Get the current block
+                let block = if let Ok(Some(block)) =
+                    self.inner.client().block_by_number_or_tag(alloy_eips::BlockNumberOrTag::Latest)
+                {
+                    block
+                } else {
+                    // Failure to get the latest block is a critical error.
+                    // Transaction validation must be re-tried.
+                    warn!(target: "reth::txpool",
+                        "Transaction validation failed: could not get latest block (interop)"
+                    );
+                    return TransactionValidationOutcome::Error(
+                        *valid_tx.hash(),
+                        InteropValidationError::Temporary.into(),
+                    )
+                };
+
+                // Construct the state using the current block.
+                let Ok(state) = self.inner.client().state_by_block_hash(block.header().hash())
+                else {
+                    warn!(target: "reth::txpool",
+                        "Transaction execution failed: failed to load state"
+                    );
+                    return TransactionValidationOutcome::Error(
+                        *valid_tx.hash(),
+                        InteropValidationError::Temporary.into(),
                     )
-                    .into(),
-                )
+                };
+                let mut db = CacheDB::new(StateProviderDatabase::new(&state));
+
+                // Configure the evm
+                let mut evm = evm_cfg.evm_for_block(&mut db, block.header());
+
+                // Construct the environment for the transaction
+                let sender = valid_tx.transaction().sender();
+                let mut tx_env = evm_cfg.tx_env(valid_tx.transaction(), sender);
+
+                // Transact
+                let logs = match evm.transact(tx_env) {
+                    Ok(ResultAndState { result, .. }) => result.logs(),
+                    Err(err) => {
+                        warn!(target: "reth::txpool",
+                            %err,
+                            "Transaction execution failed"
+                        );
+                        return TransactionValidationOutcome::Error(
+                            *valid_tx.hash(),
+                            InteropValidationError::Temporary.into(),
+                        )
+                    }
+                };
+                trace!(target: "reth::txpool",
+                    tx_hash = %valid_tx.hash(),
+                    "Tx executed successfully"
+                );
+                // todo: parse logs to extract executing message if any (sync)
+                // todo: if any executing message, return in valid variant as Some(msg)
+                // todo: from caller of this func, make async call to supervisor api.
```

Comment:
this will be handles as separate issue and depends on https://github.com/op-rs/maili/pull/129

Reply:
I've broken this up into individual tickets:
- https://github.com/ethereum-optimism/op-reth/issues/43
- https://github.com/ethereum-optimism/op-reth/issues/45